src/bufr/DataObject.h

```diff
@@ -110,9 +110,9 @@ namespace Ingester
         /// \return Float data.
         virtual float getAsFloat(const Location& loc) const = 0;
 
-        /// \brief Get the data at the index as an float.
+        /// \brief Get the data at the index as an int.
         /// \return Float data.
-        virtual float getAsFloat(size_t idx) const = 0;
+        virtual int getAsInt(size_t idx) const = 0;
```

Comment:
Ron, I am using the getAsFloat function in my feature/query_cxx_types_NE_aircraftAltitude branch. I think we should keep both

Reply:
Actually I really need to get rid of both as using this function is really dangerous if you are not careful... If you depend on any values in repeated sequences (or that could be in repeated sequences) you should not go near these methods. I'm gonna have to think about this a little I think.

Reply:
OK, I'll rework my code to not use it.

Reply:
Ok, I added it back. You need to validate the input data and make sure it all comes from the same place (same place I the tree).

Reply:
This will need to be revisited in the future.